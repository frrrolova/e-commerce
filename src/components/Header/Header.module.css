.header .logoBox:hover {
  cursor: pointer;
}
@media only screen and (max-width: 320px) {
  .header .logoBox .logoText {
    font-size: 1rem;
    line-height: 32px;
  }
}
@media only screen and (max-width: 300px) {
  .header .logoBox .logoText {
    display: none;
  }
}
.header .logo {
  height: 2rem;
  margin-right: 0.5rem;
}
.header .customIconButton {
  padding: 8px;
}
.header[data-scrolled=true] {
  background-color: #262e22;
  opacity: 0.8;
}
.header .toolbar {
  justify-content: space-between;
}
.header .navMd button {
  position: relative;
  margin-top: 21px;
}
.header .navMd button::after {
  content: "";
  position: absolute;
  bottom: 0;
  left: 0;
  width: 0;
  height: 2px;
  background-color: rgba(85, 176, 0, 0.6);
  transition: all 0.3s ease;
}
.header .activeMenuItem {
  color: rgba(85, 176, 0, 0.6);
}
.header .activeMenuItem::after {
  width: 100% !important;
}

.activeMenuItem {
  color: rgba(85, 176, 0, 0.6);
}/*# sourceMappingURL=Header.module.css.map */