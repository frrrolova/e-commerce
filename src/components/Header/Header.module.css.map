{"version":3,"sources":["Header.module.scss","Header.module.css","../../styles/variables.scss"],"names":[],"mappings":"AAII;EACE,eAAA;ACHN;ADMI;EACE;IACE,eAAA;IACA,iBAAA;ECJN;AACF;ADOI;EACE;IACE,aAAA;ECLN;AACF;ADSE;EACE,YAAA;EACA,oBAAA;ACPJ;ADUE;EACE,YAAA;ACRJ;ADWE;EACE,yBE/BU;EFgCV,YAAA;ACTJ;ADYE;EACE,8BAAA;ACVJ;ADcI;EACE,kBAAA;EACA,gBAAA;ACZN;ADcM;EACE,WAAA;EACA,kBAAA;EACA,SAAA;EACA,OAAA;EACA,QAAA;EACA,WAAA;EACA,uCE/Ca;EFgDb,yBAAA;ACZR;ADiBE;EACE,4BEtDiB;ADuCrB;ADkBE;EACE,sBAAA;AChBJ;;ADoBA;EACE,4BE/DmB;AD8CrB","file":"Header.module.css"}