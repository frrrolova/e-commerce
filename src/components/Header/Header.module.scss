@import '../../styles/variables.scss';

.header {
  .logo {
    height: 2rem;
  }

  .customIconButton {
    padding: 8px;
  }

  &[data-scrolled="true"] {
    background-color: $paper-color;
    opacity: 0.8;
  }
}

