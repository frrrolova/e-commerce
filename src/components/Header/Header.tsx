import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import Menu from '@mui/material/Menu';
import MenuIcon from '@mui/icons-material/Menu';
import LoginIcon from '@mui/icons-material/Login';
import Container from '@mui/material/Container';
import Avatar from '@mui/material/Avatar';
import Button from '@mui/material/Button';
import Tooltip from '@mui/material/Tooltip';
import MenuItem from '@mui/material/MenuItem';
import AdbIcon from '@mui/icons-material/Adb';
import { navLinksData, rightMenuData } from '../../routes/routeConstants';
import { useNavigate } from 'react-router-dom';
import styles from './Header.module.scss';
import { RootState, useAppSelector } from '../../store/store';

const pages = [navLinksData.home, navLinksData.catalog, navLinksData.about];

function Header() {
  const [anchorElNav, setAnchorElNav] = React.useState<null | HTMLElement>(null);
  const [anchorElUser, setAnchorElUser] = React.useState<null | HTMLElement>(null);

  const shopName = useAppSelector((state: RootState) => state.shop?.name?.en || 'Plant Shop');

  const navigate = useNavigate();

  // TODO take @isLogged from store and update @menuData and @currentIcon
  const isLogged = false;

  const menuData = isLogged ? rightMenuData.isAuth : rightMenuData.notAuth;

  const currentIcon = isLogged ? <Avatar alt="User" src="/static/images/avatar/2.jpg" /> : <LoginIcon />;

  const handleOpenNavMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElNav(event.currentTarget);
  };
  const handleOpenUserMenu = (event: React.MouseEvent<HTMLElement>) => {
    setAnchorElUser(event.currentTarget);
  };

  const handleCloseNavMenu = (_e: React.MouseEvent<HTMLLIElement | HTMLButtonElement, MouseEvent>, toUrl: string) => {
    if (toUrl !== 'backdropClick') {
      navigate(toUrl);
    }
    setAnchorElNav(null);
  };

  const handleCloseUserMenu = (_e: React.MouseEvent<HTMLLIElement, MouseEvent>, toUrl: string) => {
    if (toUrl !== 'backdropClick') {
      navigate(toUrl);
    }
    setAnchorElUser(null);
  };

  return (
    <AppBar position="static" className={styles.header}>
      <Container maxWidth="xl">
        <Toolbar disableGutters>
          <AdbIcon sx={{ display: { xs: 'none', md: 'flex' }, mr: 1 }} />
          <Typography
            variant="h6"
            noWrap
            sx={{
              mr: 2,
              display: { xs: 'none', md: 'flex' },
              fontFamily: 'monospace',
              fontWeight: 700,
              letterSpacing: '0.3rem',
            }}
          >
            {shopName}
          </Typography>

          <Box sx={{ flexGrow: 1, display: { xs: 'flex', md: 'none' } }}>
            <IconButton
              size="large"
              aria-label="account of current user"
              aria-controls="menu-appbar"
              aria-haspopup="true"
              onClick={handleOpenNavMenu}
              color="inherit"
            >
              <MenuIcon />
            </IconButton>
            <Menu
              id="menu-appbar"
              anchorEl={anchorElNav}
              anchorOrigin={{
                vertical: 'bottom',
                horizontal: 'left',
              }}
              keepMounted
              transformOrigin={{
                vertical: 'top',
                horizontal: 'left',
              }}
              open={Boolean(anchorElNav)}
              onClose={handleCloseNavMenu}
              sx={{
                display: { xs: 'block', md: 'none' },
              }}
            >
              {pages.map((page) => (
                <MenuItem key={page.title} onClick={(e) => handleCloseNavMenu(e, page.url)}>
                  <Typography textAlign="center">{page.title}</Typography>
                </MenuItem>
              ))}
            </Menu>
          </Box>
          <AdbIcon sx={{ display: { xs: 'flex', md: 'none' }, mr: 1 }} />
          <Typography
            variant="h5"
            noWrap
            sx={{
              mr: 2,
              display: { xs: 'flex', md: 'none' },
              flexGrow: 1,
              fontFamily: 'monospace',
              fontWeight: 700,
              letterSpacing: '0.3rem',
            }}
          >
            Plant
          </Typography>
          <Box sx={{ flexGrow: 1, display: { xs: 'none', md: 'flex' } }}>
            {pages.map((page) => (
              <Button
                key={page.title}
                onClick={(e) => handleCloseNavMenu(e, page.url)}
                sx={{ my: 2, color: 'white', display: 'block' }}
              >
                {page.title}
              </Button>
            ))}
          </Box>

          <Box sx={{ flexGrow: 0 }}>
            <Tooltip title={menuData.tooltipTitle}>
              <IconButton
                onClick={handleOpenUserMenu}
                className={styles.customIconButton}
                size="large"
                sx={{ p: 0 }}
                aria-label="login"
              >
                {currentIcon}
              </IconButton>
            </Tooltip>
            <Menu
              sx={{ mt: '45px' }}
              id="menu-appbar"
              anchorEl={anchorElUser}
              anchorOrigin={{
                vertical: 'top',
                horizontal: 'right',
              }}
              keepMounted
              transformOrigin={{
                vertical: 'top',
                horizontal: 'right',
              }}
              open={Boolean(anchorElUser)}
              onClose={handleCloseUserMenu}
            >
              {menuData.links.map((link) => (
                <MenuItem key={link.title} onClick={(e) => handleCloseUserMenu(e, link.url)}>
                  <Typography textAlign="center">{link.title}</Typography>
                </MenuItem>
              ))}
            </Menu>
          </Box>
        </Toolbar>
      </Container>
    </AppBar>
  );
}
export default Header;
