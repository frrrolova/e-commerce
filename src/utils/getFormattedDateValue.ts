export function getFormattedDateValue(value: string) {
  const date = new Date(value);
  const dateValue: string = new Date(date.getTime() - date.getTimezoneOffset() * 60000).toISOString().substring(0, 10);
  return dateValue;
}
