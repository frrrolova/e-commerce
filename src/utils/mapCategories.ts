import { Category } from '@/types';
import { Category as CategorySDK } from '@commercetools/platform-sdk';

export function mapCategories(categoryResponse: CategorySDK[]): Category[] {
  return categoryResponse.map((category) => ({
    id: category.id,
    name: category.name['en-US'],
    key: category.key,
    ancestors: category.ancestors,
    parent: category.parent,
  }));
}
