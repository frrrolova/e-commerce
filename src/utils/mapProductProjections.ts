import { Product } from '@/types';
import { ProductProjection } from '@commercetools/platform-sdk';

export function mapProductProjections(projections: ProductProjection[]): Product[] {
  return projections.map((projection) => ({
    id: projection.id,
    name: projection.name['en-GB'],
    description: projection.description?.['en-GB'] || 'Product has no description',
    images: projection.masterVariant.images,
    prices: projection.masterVariant.prices,
  }));
}
