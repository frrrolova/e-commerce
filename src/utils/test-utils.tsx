import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';

import type { AppStore, RootState } from '../store/store';
// As a basic setup, import your same slice reducers
import initStore from '../store/store';
import { MemoryRouter } from 'react-router-dom';
import { LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import { Project } from '@commercetools/platform-sdk';

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

const mockShop: Project = {
  countries: [],
  createdAt: new Date().toISOString(),
  languages: [],
  key: 'test-key',
  version: 2,
  name: 'Plant Shop',
  currencies: [],
  messages: { enabled: false, deleteDaysAfterCreation: 15 },
  carts: {
    deleteDaysAfterLastModification: 90,
    countryTaxRateFallbackEnabled: false,
  },
};

export function renderWithProviders(ui: React.ReactElement, extendedRenderOptions: ExtendedRenderOptions = {}) {
  const {
    preloadedState = {
      project: mockShop,
    },
    // Automatically create a store instance if no store was passed in
    store = initStore(preloadedState.project as Project),
    ...renderOptions
  } = extendedRenderOptions;

  const Wrapper = ({ children }: PropsWithChildren) => (
    <MemoryRouter>
      <LocalizationProvider dateAdapter={AdapterDayjs}>
        <Provider store={store}>{children}</Provider>
      </LocalizationProvider>
    </MemoryRouter>
  );

  // Return an object with the store and all of RTL's query functions
  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
  };
}
