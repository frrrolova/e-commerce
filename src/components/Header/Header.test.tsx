import '@testing-library/jest-dom';
import Header from './Header';
import { renderWithProviders } from '../../utils/test-utils';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

jest.mock('../../client/client', () => {
  return null;
});

//Mock for useNavigate
const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('Header component rendering', () => {
  test('renders Header Title', () => {
    const { getByTestId } = renderWithProviders(<Header />);
    const shopNameEl = getByTestId('shop-name');
    expect(shopNameEl.textContent?.trim()).toBe('Plant Shop');
  });

  test('performs snapshot testing', () => {
    const tree = renderWithProviders(<Header />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Header component behavior', () => {
  let homeMenuItem: Node | Window;

  test('opens and closes navigation left-menu when clicking on menu icon', async () => {
    const { getByLabelText, getByTestId } = render(
      <MemoryRouter>
        <Header />
      </MemoryRouter>,
    );

    const menuIcon = getByLabelText('account of current user');
    fireEvent.click(menuIcon);

    homeMenuItem = getByTestId('menu-item-Home');
    const menu = getByTestId('menu-left');
    expect(homeMenuItem).toBeInTheDocument();

    fireEvent.click(homeMenuItem);

    await waitFor(() => {
      expect(menu).toHaveStyle('visibility: hidden;');
    });
  });

  test('should navigate, when homeMenuItem is clicked', () => {
    fireEvent.click(homeMenuItem);
    expect(mockNavigate).toHaveBeenCalledTimes(1);
  });
});
