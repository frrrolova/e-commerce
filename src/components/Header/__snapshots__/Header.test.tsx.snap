// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component rendering performs snapshot testing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
        data-scrolled="false"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
          >
            <div
              class="MuiBox-root css-1guk29"
            >
              <img
                alt="Logo"
                loading="lazy"
                src="default"
              />
              <h6
                class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-6qa9vu-MuiTypography-root"
                data-testid="shop-name"
              >
                Plant Shop
              </h6>
            </div>
            <div
              class="MuiBox-root css-1lvtzne"
            >
              <button
                aria-controls="menu-appbar"
                aria-haspopup="true"
                aria-label="account of current user"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MenuIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiBox-root css-1k36coi"
            >
              <img
                alt="Logo"
                loading="lazy"
                src="default"
              />
            </div>
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap css-wqc8kw-MuiTypography-root"
            >
              Plant
            </h5>
            <div
              class="MuiBox-root css-1guk29"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1q39md6-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Home
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1q39md6-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Catalog
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1q39md6-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                About us
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiBox-root css-2uchni"
            >
              <button
                aria-label="login"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-16rd89w-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LoginIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </header>
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root MuiMenu-root MuiModal-root MuiModal-hidden css-10nezku-MuiModal-root-MuiPopover-root-MuiMenu-root"
      data-testid="menu-left"
      id="menu-appbar"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="-1"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper MuiMenu-paper MuiMenu-paper css-3dzjca-MuiPaper-root-MuiPopover-paper-MuiMenu-paper"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
              data-testid="menu-item-Home"
            >
              Home
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
              data-testid="menu-item-Catalog"
            >
              Catalog
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
              data-testid="menu-item-About us"
            >
              About us
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="-1"
      />
    </div>
    <div
      aria-hidden="true"
      class="MuiPopover-root MuiMenu-root MuiModal-root MuiModal-hidden css-14ogsoa-MuiModal-root-MuiPopover-root-MuiMenu-root"
      id="menu-appbar"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="-1"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper MuiMenu-paper MuiMenu-paper css-3dzjca-MuiPaper-root-MuiPopover-paper-MuiMenu-paper"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
        tabindex="-1"
      >
        <ul
          class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
          role="menu"
          tabindex="-1"
        >
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
            >
              Login
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
          <li
            class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
            role="menuitem"
            tabindex="-1"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
            >
              Registration
            </p>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </li>
        </ul>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="-1"
      />
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      data-scrolled="false"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
        >
          <div
            class="MuiBox-root css-1guk29"
          >
            <img
              alt="Logo"
              loading="lazy"
              src="default"
            />
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-6qa9vu-MuiTypography-root"
              data-testid="shop-name"
            >
              Plant Shop
            </h6>
          </div>
          <div
            class="MuiBox-root css-1lvtzne"
          >
            <button
              aria-controls="menu-appbar"
              aria-haspopup="true"
              aria-label="account of current user"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiBox-root css-1k36coi"
          >
            <img
              alt="Logo"
              loading="lazy"
              src="default"
            />
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap css-wqc8kw-MuiTypography-root"
          >
            Plant
          </h5>
          <div
            class="MuiBox-root css-1guk29"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1q39md6-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Home
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1q39md6-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Catalog
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1q39md6-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              About us
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiBox-root css-2uchni"
          >
            <button
              aria-label="login"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-16rd89w-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LoginIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
